### PR TITLE
fix: remove redundant postgres indexes

### LINKED ISSUE #963: remove redundant postgres indexes
PgHero reports these indexes could be removed for faster writes:

```
On contract_logs
contract_logs_canonical_index (canonical)
is covered by
contract_logs_canonical_microblock_canonical_index (canonical, microblock_canonical)

On ft_events
ft_events_canonical_index (canonical)
is covered by
ft_events_canonical_microblock_canonical_index (canonical, microblock_canonical)

On mempool_txs
mempool_txs_tx_id_index (tx_id)
is covered by
unique_tx_id (tx_id)

On namespaces
namespaces_namespace_id_index (namespace_id)
is covered by
namespaces_namespace_id_canonical_microblock_canonical_ready_bl (namespace_id, canonical DESC, microblock_canonical DESC, ready_block DESC, tx_index DESC)

On nft_events
nft_events_asset_identifier_index (asset_identifier)
is covered by
nft_events_asset_identifier_value_index (asset_identifier, value)

On nft_events
nft_events_canonical_index (canonical)
is covered by
nft_events_canonical_microblock_canonical_index (canonical, microblock_canonical)

On stx_events
stx_events_canonical_index (canonical)
is covered by
stx_events_canonical_microblock_canonical_index (canonical, microblock_canonical)

On stx_events
stx_events_recipient_index (recipient)
is covered by
stx_events_recipient_block_height_index (recipient, block_height)

On stx_events
stx_events_sender_index (sender)
is covered by
stx_events_sender_block_height_index (sender, block_height)

On stx_lock_events
stx_lock_events_index_block_hash_index (index_block_hash)
is covered by
stx_lock_events_index_block_hash_tx_id_index (index_block_hash, tx_id)

On subdomains
subdomains_canonical_index (canonical)
is covered by
subdomains_canonical_microblock_canonical_index (canonical, microblock_canonical)

On subdomains
subdomains_fully_qualified_subdomain_index (fully_qualified_subdomain)
is covered by
subdomains_fully_qualified_subdomain_canonical_microblock_canon (fully_qualified_subdomain, canonical DESC, microblock_canonical DESC, block_height DESC, tx_index DESC)

On txs
txs_canonical_index (canonical)
is covered by
txs_canonical_microblock_canonical_index (canonical, microblock_canonical)

On txs
txs_sender_address_index (sender_address)
is covered by
txs_sender_address_block_height_index (sender_address, block_height)

On txs
txs_tx_id_index (tx_id)
is covered by
unique_tx_id_index_block_hash_microblock_hash (tx_id, index_block_hash, microblock_hash)
```

### PR DESCRIPTION
**Pull request by rafaelcr**
Closes #963

### CODE REVIEW

**Review by CharlieC3 — APPROVED** [review]
Would this require an event-replay to ensure the indices don't get created?

**Comment by rafaelcr** [comment]
Yeah, I believe so @CharlieC3 ... we could ship this with the next release which will definitely need a replay

File: src/migrations/1584619633448_txs.ts [hunk 1]
```diff
@@ -157,26 +157,21 @@ export async function up(pgm: MigrationBuilder): Promise<void> {
 
   });
 
-  pgm.createIndex('txs', 'tx_id');
```
Comment (zone117x) [inline src/migrations/1584619633448_txs.ts:160]: This one is surprising. There's several queries against the `txs` table using the `tx_id` column as a primary search condition. I can definitely see how removing the index would speed up writes, but seems like this one would dramatically slow slow down a bunch of reads?

Comment (rafaelcr) [reply src/migrations/1584619633448_txs.ts:160]: I actually validated this using `EXPLAIN ANALYZE`, and that index is just substituted by the `unique_tx_id_index_block_hash_microblock_hash` index when querying by `tx_id`, example:
```
Index Scan using unique_tx_id_index_block_hash_microblock_hash on txs  (cost=0.55..8.57 rows=1 width=1585) (actual time=2.181..2.193 rows=1 loops=1)
"  Index Cond: (tx_id = '\x4510e6cff6414747f402126b92e35d90165f9f5e1f2750f50ed2b8e052b25108'::bytea)"
Planning Time: 0.159 ms
Execution Time: 2.239 ms
```

Comment (zone117x) [reply src/migrations/1584619633448_txs.ts:160]: That's cool! I wasn't aware constraints could be used as indexes like that, makes sense though!

**Review by zone117x — APPROVED** [review]

**Pull request merged**